File: csci40/lec02/counter.cpp
#include <iostream>
using namespace std;

class Counter {
 public:
  int count;

  void reset() { count = 0; }

  void inc() { count = count + 1; }
};

int main() {
  Counter c;  // make a brand new Counter called c

  c.reset();  // reset the counter's count to 0
  c.inc();
  c.inc();
  c.inc();

  cout << "Current count: " << c.count << endl;

  c.reset();
  c.inc();
  c.inc();

  cout << "Current count: " << c.count << endl;

  return 0;
}

File: csci40/lec31/counter.py
class Counter:
    # constructor is called __init__
    # every method takes an explicit "self" parameter first
    def __init__(self, count=0):
        self.count = count # makes the member var right here!

    def inc(self):
        self.count += 1

    def getCount(self):
        return self.count

# notice that at the time of use for each method, self gets passed for
# you
c = Counter()
c.inc()
c.inc()
c.inc()
print(c.getCount())
c2 = Counter(42)
print(c2.getCount())
